pkgs/tools/networking/zrok/default.nix

```diff
@@ -1,15 +1,37 @@
 { stdenv, lib, fetchzip, patchelf }:
 
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  plat = {
+    x86_64-linux = "linux_amd64";
+    x86_64-darwin = "darwin_amd64";
+    aarch64-linux = "linux_arm64";
+    aarch64-darwin = "darwin_arm64";
+    armv7l-linux = "linux_armv7";
+  }.${system} or throwSystem;
+
+  sha256 = {
+    x86_64-linux = "sha256-gcmgpvfk7bciTmotTHObvZvLPdLudAR2vQneLKN+uE4=";
+    x86_64-darwin = "sha256-YbMsKfFHyCZjmKyK4YMm5lpLluvrwsJ2amAxbALCKus=";
+    aarch64-linux = "sha256-cZDekIW2Z7hfIY6Y4xhsmgvMLnKYo6H9BAMg9/I5a10=";
+    aarch64-darwin = "sha256-YbMsKfFHyCZjmKyK4YMm5lpLluvrwsJ2amAxbALCKus=";
+    armv7l-linux = "sha256-Fn2e1ywuTUt58geT4hNb3YEZiJDE6TUKaKI/xpsG69c=";
+  }.${system} or throwSystem;
+in
 stdenv.mkDerivation rec {
   pname = "zrok";
   version = "0.3.6";
 
   src = fetchzip {
-    url = "https://github.com/openziti/zrok/releases/download/v${version}/zrok_${version}_linux_amd64.tar.gz";
+    url = "https://github.com/openziti/zrok/releases/download/v${version}/zrok_${version}_${plat}.tar.gz";
     stripRoot = false;
-    sha256 = "sha256-gcmgpvfk7bciTmotTHObvZvLPdLudAR2vQneLKN+uE4=";
+    inherit sha256;
   };
 
+  updateScript = ./update.sh;
+
   installPhase = let
     interpreter = "$(< \"$NIX_CC/nix-support/dynamic-linker\")";
   in ''
```

Comment:
```
building '/nix/store/sgi21wc87b1n455zfi9dgm41pmlkfaiq-zrok-0.3.6.drv'...
unpacking sources
unpacking source archive /nix/store/gyfwhj5ma0dz61fsbmkmzv89s20y2jrx-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
no Makefile or custom buildPhase, doing nothing
installing
/nix/store/r46ydv8hlmx68nnclchjbgq0gjv6nr9n-stdenv-darwin/setup: line 1610: patchelf: command not found
error: builder for '/nix/store/sgi21wc87b1n455zfi9dgm41pmlkfaiq-zrok-0.3.6.drv' failed with exit code 127;
```

Reply:
Thanks. I dropped the darwin stuff since I don't have the hardware to test it.